internal/server/data/grant.go

```diff
@@ -1,16 +1,37 @@
 package data
 
 import (
+	"errors"
 	"fmt"
 
+	"github.com/jackc/pgconn"
+	"github.com/jackc/pgerrcode"
 	"gorm.io/gorm"
 
 	"github.com/infrahq/infra/internal/logging"
 	"github.com/infrahq/infra/internal/server/models"
 	"github.com/infrahq/infra/uid"
 )
 
-func CreateGrant(db GormTxn, grant *models.Grant) error {
+type grantsTable models.Grant
+
+func (g grantsTable) Table() string {
+	return "grants"
+}
+
+func (g grantsTable) Columns() []string {
+	return []string{"created_at", "created_by", "deleted_at", "id", "organization_id", "privilege", "resource", "subject", "updated_at"}
+}
+
+func (g grantsTable) Values() []any {
+	return []any{g.CreatedAt, g.CreatedBy, g.DeletedAt, g.ID, g.OrganizationID, g.Privilege, g.Resource, g.Subject, g.UpdatedAt}
+}
+
+func (g *grantsTable) ScanFields() []any {
+	return []any{&g.CreatedAt, &g.CreatedBy, &g.DeletedAt, &g.ID, &g.OrganizationID, &g.Privilege, &g.Resource, &g.Subject, &g.UpdatedAt}
+}
```

Comment:
seems like we could find a better way to do this. Is there reason why it's a composite currently?

Reply:
Sorry, I don't follow. What do you mean by composite?

Reply:
We talked about this a bit offline. We could move these methods to the model struct, but then everything would have access to them. By creating a new type using the same data we can hide these methods and only expose them to the data package.